src/GraphRequest.ts

```diff
@@ -501,19 +494,29 @@ export class GraphRequest {
 	 * @returns The same GraphRequest instance that is being called with
 	 */
 	public query(queryDictionaryOrString: string | KeyValuePairObjectStringNumber): GraphRequest {
-		const otherURLQueryParams = this.urlComponents.otherURLQueryParams;
+		let paramKey: string;
+		let paramValue: string | number;
 		if (typeof queryDictionaryOrString === "string") {
-			const querySplit = queryDictionaryOrString.split("=");
-			const queryKey = querySplit[0];
-			const queryValue = querySplit[1];
-			otherURLQueryParams[queryKey] = queryValue;
+			/* The query key-value pair must be split on the first equals sign to avoid errors in parsing nested query parameters.
+			 Example-> "/me?$expand=home($select=city)" */
+			const indexOfFirstEquals = queryDictionaryOrString.indexOf("=");
+			paramKey = queryDictionaryOrString.substring(0, indexOfFirstEquals);
+			paramValue = queryDictionaryOrString.substring(indexOfFirstEquals + 1, queryDictionaryOrString.length);
 		} else {
 			for (const key in queryDictionaryOrString) {
 				if (queryDictionaryOrString.hasOwnProperty(key)) {
-					otherURLQueryParams[key] = queryDictionaryOrString[key];
+					paramKey = key;
+					paramValue = queryDictionaryOrString[key];
```

Comment:
shouldn't there be a break here? if not can you add a comment about the intent of the code here.  once paramKey is set it's not clear why continue

Reply:
It was a mistake on my part. Corrected the code so that the iteration records key and value for each parameter in the dictionary.